UOP1_Project/Assets/Scripts/StateMachine/Core/StateAction.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace DeivSky.StateMachine
+{
+	/// <summary>
+	/// An object representing an action.
+	/// </summary>
+	public abstract class StateAction : IStateComponent
+	{
+		/// <summary>
+		/// Perform the action this object represents.
+		/// </summary>
+		public abstract void Perform();
```

Comment:
I used that name because at first I didn't have `OnStateEnter` and `OnStateExit` but rather three sets of `StateAction[]` in the `State`, them being `_onStateEnterActions`, `_onStateExitActions`, and `_onUpdateActions` that the `State` would call in their respective times. So I needed a generic name for it. When I switched formats I just kept the name. I'll change it to `OnUpdate`.